cli/src/lib.rs

```diff
@@ -570,6 +556,25 @@ fn new(cfg_override: &ConfigOverride, name: String) -> Result<()> {
     })
 }
 
+fn rename(cfg_override: &ConfigOverride, from: String, to: String) -> Result<()> {
+    is_name_valid(&to)?;
+    with_workspace(cfg_override, |cfg| {
+        match cfg.path().parent() {
+            None => {
+                println!("Unable to rename workspace");
+            }
+            Some(parent) => {
```

Comment:
how about instead of hardcoding locations here, you just iterate through all directories (except those in `.gitignore`) to change file/dirnames and contents

Reply:
yeah I suppose we can't assume workspace structure here. i'll add it.

Reply:
i've integrated the `WalkBuilder` crate which traverses the tree, automatically skipping .gitignore entries. it's working well.